### PR TITLE
Adding tests and documentation

### PR DESCRIPTION
**Pull request by aiday-mar**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,43 +1,83 @@
...
+          kind: string;
+        };
+    /**
+     * the kind and options of the algorithm use to optimize parameters
+     */
+    optimization?: {
+      /**
+       * kind of algorithm. By default it's levenberg-marquardt
+       */
+      kind?: string | number;
```
Comment (lpatiny) [inline src/index.ts:38]: Why should we allow a number ?

File: src/index.ts [hunk 2]
```diff
@@ -1,43 +1,83 @@
...
   data: DataXY<DoubleArray>,
   peakList: Peak1D[],
-  options: OptimizeOptions,
+  options: {
+    /**
+     * kind of shape used for fitting
+     **/
+    shape?:
+      | Shape1D
+      | {
```
Comment (lpatiny) [inline src/index.ts:25]: I think we should only allow a Shape1D

File: src/index.ts [hunk 3]
```diff
@@ -1,43 +1,83 @@
...
+ * @returns - An object with fitting error and the list of optimized parameters { parameters: [ {x, y, width} ], error } if the kind of shape is pseudoVoigt mu parameter is optimized.
  */
-
 export function optimize(
   data: DataXY<DoubleArray>,
   peakList: Peak1D[],
-  options: OptimizeOptions,
+  options: {
+    /**
+     * kind of shape used for fitting
```
Comment (lpatiny) [inline src/index.ts:21]: Please don't forget to add the @default if there is any.

File: src/shapes/sumOfLorentzians.ts [hunk 4]
```diff
@@ -3,18 +3,16 @@ import { Lorentzian } from 'ml-peak-shape-generator';
 /**
  * This function calculates the spectrum as a sum of lorentzian functions. The Lorentzian
  * parameters are divided in 3 batches. 1st: centers; 2nd: heights; 3th: widths;
- * @param parameters Lorentzian parameters
+ * @param parameters - Lorentzian parameters
  * @returns {*}
```
Comment (lpatiny) [inline src/shapes/sumOfLorentzians.ts:7]: Please have a look at all the @returns and remove the one that are useless.

File: src/util/checkInput.ts [hunk 5]
```diff
@@ -26,13 +31,23 @@ export function checkInput(
...
 
+  // removing the letters by nothing
   let kind = shape.kind.toLowerCase().replace(/[^a-z]/g, '');
 
   let paramsFunc;
   let defaultParameters;
   switch (kind) {
     case 'gaussian':
       paramsFunc = sumOfGaussians;
+      /*
```
Comment (lpatiny) [inline src/util/checkInput.ts:42]: Why did you add this comment here ? Not sure it helps.

File: src/util/selectMethod.ts [hunk 6]
```diff
@@ -4,13 +4,18 @@ import { OptimizationOptions } from '../spectra-fitting';
...
+ * @returns - The algorithm and optimization options
+ */
 export function selectMethod(optimizationOptions: OptimizationOptions = {}) {
   let { kind, options } = optimizationOptions;
   kind = getKind(kind);
   switch (kind) {
     case LEVENBERG_MARQUARDT:
       return {
         algorithm: levenbergMarquardt,
+        // check the options
```
Comment (lpatiny) [inline src/util/selectMethod.ts:18]: Please review the added comments because when a method is called `checkOptions` we should not add a comment telling it `check the options`.

If the function name requires a comment on the other hand maybe we should rename the function.

**Review by lpatiny — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (aiday-mar) [reply src/index.ts:25]: In the test file though we have the following though:

optimize(modifiedData, peakList, {
      shape: { kind: 'lorentzian' }
});
